test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -317,3 +317,480 @@ test.describe( 'Navigation block', () => {
+		await firstMenuItemAnchor.focus();
+
+		// The options menu button is a sibling of the menu item gridcell.
+		const firstItemOptions = firstMenuItem
+			.locator( '..' ) // parent selector.
+			.getByRole( 'button', {
+				name: 'Options for Page Link block',
+			} );
+
+		// Open the options menu.
+		await firstItemOptions.click();
+
+		// usage of `page` is required because the options menu is rendered into a slot
+		// outside of the treegrid.
+		const removeBlockOption = page
+			.getByRole( 'menu', {
+				name: 'Options for Page Link block',
+			} )
+			.getByRole( 'menuitem', {
+				name: 'Remove Top Level Item 1',
+			} );
+
+		await removeBlockOption.click();
+
+		// Check the menu item was removed.
+		await expect(
+			listView
+				.getByRole( 'gridcell', {
+					name: 'Page Link link',
+				} )
+				.filter( {
+					hasText: 'Block 1 of 1, Level 1', // proxy for filtering by description.
+				} )
+				.getByText( 'Top Level Item 1' )
+		).not.toBeVisible();
+	} );
+
+	test( `can edit menu items`, async ( {
+		admin,
+		page,
+		editor,
+		requestUtils,
+	} ) => {
+		await admin.createNewPost();
+		await requestUtils.createNavigationMenu( {
+			title: 'Test Menu',
+			content: `<!-- wp:navigation-link {"label":"Top Level Item 1","type":"page","id":250,"url":"http://localhost:8888/quod-error-esse-nemo-corporis-rerum-repellendus/","kind":"post-type"} /-->
+			<!-- wp:navigation-submenu {"label":"Top Level Item 2","type":"page","id":250,"url":"http://localhost:8888/quod-error-esse-nemo-corporis-rerum-repellendus/","kind":"post-type"} -->
+				<!-- wp:navigation-link {"label":"Test Submenu Item","type":"page","id":270,"url":"http://localhost:8888/et-aspernatur-recusandae-non-sint/","kind":"post-type"} /-->
+			<!-- /wp:navigation-submenu -->`,
+		} );
+
+		await editor.insertBlock( { name: 'core/navigation' } );
+
+		await editor.openDocumentSettingsSidebar();
+
+		const listViewTab = page.getByRole( 'tab', {
+			name: 'List View',
+		} );
+
+		await listViewTab.click();
+
+		const listView = page.getByRole( 'treegrid', {
+			name: 'Block navigation structure',
+			description: 'Structure for navigation menu: Test Menu',
+		} );
+
+		// Click on the first menu item to open its settings.
+		const firstMenuItemAnchor = listView.getByRole( 'link', {
+			name: 'Top Level Item 1',
+			includeHidden: true,
+		} );
+		await firstMenuItemAnchor.click();
+
+		// Get the settings panel.
+		const blockSettings = page.getByRole( 'region', {
+			name: 'Editor settings',
+		} );
+
+		await expect( blockSettings ).toBeVisible();
+
+		await expect(
+			blockSettings.getByRole( 'heading', {
+				name: 'Page Link',
+			} )
+		).toBeVisible();
+
+		await expect(
+			blockSettings.getByRole( 'tab', {
+				name: 'Settings',
+				selected: true,
+			} )
+		).toBeVisible();
+
+		await expect(
+			blockSettings
+				.getByRole( 'tabpanel', {
+					name: 'Settings',
+				} )
+				.getByRole( 'heading', {
+					name: 'Link Settings',
+				} )
+		).toBeVisible();
```

Comment:
It might be good to edit the label in this test, and then check that the change is reflected in List View.

Reply:
Yes actually that's a good point. That is something that needs to be asserted on.

Reply:
Updated the test to amend the label and assert on the change taking effect.